R/shapley.R

```diff
@@ -83,10 +83,8 @@ shapr <- function(x,
   # Checks model and features
   explainer$p <- predict_model(model, head(x))
 
-  # Create data.table --------------
-  if (!data.table::is.data.table(x)) {
-    x_train <- data.table::as.data.table(x)
-  }
+  # Converts to data.table, otherwise copy to x_train  --------------
+  x_train <- data.table::as.data.table(x)
 
```

Comment:
Nice catch!

Could we instead add an else statement like this? 
```
# Create data.table --------------
if (!data.table::is.data.table(x)) {
    x_train <- data.table::as.data.table(x)
} else {
    x_train <- data.table::copy(x)
}
```

Reply:
I actually did that first, then realized that as.data.table.data.table(x) does almost exactly this, so wouldn't my suggestion just simplify the code?

https://github.com/Rdatatable/data.table/blob/50f815780297aaea5e30e9fe3c787d931957fe21/R/as.data.table.R#L239-L248

Reply:
Awesome. Didn't know that. Go for it!